File: cases/taint/typescript/expression-positive/ExpressionPositive.ts
function dfb_source(): number { // DFB-SOURCE: expression-input
  return 4;
}

function dfb_sink(value: number): void {} // DFB-SINK: expression-sink

function run(): void {
  const value: number = dfb_source();
  const computed: number = (value * 3) + 7; // DFB-WITNESS: expression-computed
  dfb_sink(computed);
}

File: cases/taint/scala/expression-positive/ExpressionPositive.scala
package dataflowbench

object ExpressionPositive {
  def dfb_source(): Int = { // DFB-SOURCE: expression-input
    1
  }

  def dfb_sink(value: Int): Unit = {} // DFB-SINK: expression-sink

  def run(): Unit = {
    val value = dfb_source()
    val computed = (value * 3) + 7 // DFB-WITNESS: expression-computed
    dfb_sink(computed)
  }
}
